src/sql/workbench/api/common/notebooks/vscodeExecuteProvider.ts

```diff
@@ -71,28 +72,30 @@ class VSCodeKernel implements azdata.nb.IKernel {
 	private readonly _info: azdata.nb.IInfoReply;
 	private readonly _kernelSpec: azdata.nb.IKernelSpec;
 
-	constructor(private readonly _controller: ADSNotebookController, private readonly _options: azdata.nb.ISessionOptions, language: string) {
+	constructor(private readonly _controller: ADSNotebookController, private readonly _options: azdata.nb.ISessionOptions, languages: string[]) {
 		this._id = this._options.kernelId ?? (VSCodeKernel.kernelId++).toString();
-		this._name = this._options.kernelName ?? this._controller.notebookType;
+		this._kernelSpec = this._options.kernelSpec ?? {
+			name: this._controller.notebookType,
+			language: languages[0],
+			display_name: this._controller.label,
+			supportedLanguages: languages
+		};
+
+		this._name = this._kernelSpec.name;
 		this._info = {
 			protocol_version: '',
 			implementation: '',
 			implementation_version: '',
 			language_info: {
-				name: language,
-				version: '',
+				name: this._kernelSpec.language,
+				supportedLanguages: languages
```

Comment:
Why is version being removed?

Reply:
It's an optional field, so this line wasn't needed.